system/jhtree/jhtree.cpp

```diff
@@ -3382,6 +3388,30 @@ class CKeyMerger : public CKeyLevelManager
         {
             if (ctx)
                 ctx->noteStatistic(StNumIndexMerges, activekeys);
+            if (traceKeyMergeThreshold && activekeys > traceKeyMergeThreshold)
+            {
+                unsigned now = msTick();
+                if (now - lastMergeThresholdReport > 60000)
+                {
+                    if (mergeThresholdExceededSuppressed)
+                    {
+                        // Report skipped
+                        DBGLOG("%u large key mergers detected with up to %u active keys", mergeThresholdExceededSuppressed.load(), maxActiveKeys.load());
+                        lastMergeThresholdReport = now;
+                        mergeThresholdExceededSuppressed = 0;
+                        maxActiveKeys = 0;
+                    }
+                    // Report this
+                    DBGLOG("Large key merger detected with %u active keys", activekeys);
```

Comment:
I might have combined or elsed this log msg

Reply:
I think I agree, I would make the previous logging unconditional, and move the maxActiveKey update on line 3411 before the now/msTick() check